lib/vanagon/utilities.rb

```diff
@@ -281,5 +281,16 @@ def erb_file(erbfile, outfile = nil, remove_orig = false, opts = { :binding => b
       FileUtils.rm_rf erbfile if remove_orig
       outfile
     end
+
+    # Final
+    # - 1.0.0
+    # - 3.0.5.1
+    # - 4.3.0-5
+    # Nonfinal
+    # - 1.0.0-34-g1sadjf9283
+    # - 4.2.1.2344.gfdqw83rh
+    def final_version?(version)
+      !version.match(/g[a-f0-9]{7}$/)
```

Comment:
This isn't just for git sources, it's for the project as a whole. If the package is versioned from the git tag, then this matches, and if it's versioned from a string, I would assume it's always final (i.e., puppet-enterprise-vanagon projects). There reason I checked for the git describe is that we always have this in the version because it's in the automation. However, if we're building something from a tag or from a specific version (i.e., `proj.version '1.0.0'`, we have no control over what tag or version the user is going to specify. This could be `postgres-2017.4.0`, `1.0.0-1`, `1.0.0-rc0`, `random-string`, anything really. I wanted to keep this as simple as possible rather than trying to predict what crazy version strings we might encounter in the future.